--- fs/dcache.c
+++ fs/dcache.c
@@ -2213,6 +2213,13 @@ struct dentry *__d_lookup_rcu(const struct dentry *parent,
 		if (dentry_cmp(dentry, str, hashlen_len(hashlen)) != 0)
 			continue;
+#ifdef CONFIG_KSU_SUSFS_SUS_PATH
+		if (dentry->d_inode &&
+		    unlikely(dentry->d_inode->i_state & INODE_STATE_SUS_PATH) &&
+		    likely(current->susfs_task_state & TASK_STRUCT_NON_ROOT_USER_APP_PROC)) {
+			continue;
+		}
+#endif
 		*seqp = seq;
 		return dentry;
