--- include/linux/sched.h
+++ include/linux/sched.h
@@ -1498,8 +1498,15 @@ struct task_struct {
 	ANDROID_KABI_RESERVE(3);
 	ANDROID_KABI_RESERVE(4);
 	ANDROID_KABI_RESERVE(5);
-	ANDROID_KABI_RESERVE(6);
-	ANDROID_KABI_RESERVE(7);
-	ANDROID_KABI_RESERVE(8);
+
+#if defined(CONFIG_KSU_SUSFS)
+	ANDROID_KABI_USE(6, u64 susfs_task_state);
+	ANDROID_KABI_USE(8, u64 susfs_last_fake_mnt_id);
+#else
+	ANDROID_KABI_RESERVE(6);
+	ANDROID_KABI_RESERVE(8);
+#endif
+
+	ANDROID_KABI_RESERVE(7);
 
 	/*
 	 * New fields for task_struct should be added above here, so that
 	 * they are included in the randomized portion of task_struct.
 	 */
 	randomized_struct_fields_end
 
 	/* CPU-specific state of this task: */
