--- fs/internal.h.orig	2025-07-10 22:07:03.003318559 +0000
+++ fs/internal.h	2025-07-10 22:17:45.598142408 +0000
@@ -78,6 +78,15 @@
 
 extern void __init mnt_init(void);
 
+#ifdef CONFIG_KSU_SUSFS
+struct umount_opts {
+    int flags;
+};
+#endif
+
+int path_umount(struct path *path, int flags);
+
+
 extern int __mnt_want_write_file(struct file *);
 extern void __mnt_drop_write_file(struct file *);
 
